code/GatewayFieldsFactory.php

```diff
@@ -320,47 +340,118 @@ protected function cullForGateway(&$fields, $defaults = array())
+     * @return mixed
+     */
+    public function getGlobalPrefix()
+    {
+        return $this->getGlobalFieldName('prefix');
+    }
+
+    /**
+     * Fetches custom name from the rename map, or returns false
+     *
+     * @param string $defaultName The original name of the field
+     *
+     * @return bool|string Returns false if no custom name has been defined, otherwise returns the custom name
+     */
+    public function getGlobalFieldName($defaultName)
+    {
+        $renameMap = $this->getRenameConfig();
+
+        if (is_array($renameMap) && array_key_exists($defaultName, $renameMap)) {
+            return $renameMap[ $defaultName ];
         }
 
-        $stack = array(array_flip($globalMap));
-        
-        if (array_key_exists($this->gateway, $renameMap) && !empty($renameMap[$this->gateway])) {
-            $stack[] = $gatewayMap = array_flip($renameMap[$this->gateway]);
+        return false;
+    }
+
+    /**
+     * Fetches the prefix for a gateway, or returns false
+     *
+     * @param null $gateway
+     *
+     * @return bool|string Returns false if no gateway prefix found, otherwise returns the prefix
+     */
+    public function getGatewayPrefix($gateway = null)
+    {
+        return $this->getGatewayFieldName('prefix', $gateway);
+    }
+
+    /**
+     * Fetches custom name for a gateway field from the rename map, or returns false
+     *
+     * @param      $defaultName
+     * @param null $gateway Optional, will default to current gateway if instantiated
+     *
+     * @return bool|string Returns false if no custom gateway field name has been defined, otherwise returns the custom
+     *                     name
+     */
+    public function getGatewayFieldName($defaultName, $gateway = null)
+    {
+        if (!$gateway) {
+            if (!$this->gateway) {
+                return false;
+            }
+
+            return $this->getGatewayFieldName($defaultName, $this->gateway);
         }
 
-        foreach($stack as $map) {
-            foreach ($map as $otherName => $defaultName) {
-                if ($hasPrefix) {
-                    $otherName = $renameMap['prefix'] . $otherName;
-                }
-                if (array_key_exists($otherName, $data)) {
-                    $value = $data[$otherName];
-                    unset($data[$otherName]);
-                    $data[$defaultName] = $value;
-                }
+        $renameMap = $this->getRenameConfig();
+
+        if (is_array($renameMap) && array_key_exists($gateway, $renameMap)) {
+            $gatewayMap = $renameMap[ $gateway ];
+            if (array_key_exists($defaultName, $gatewayMap)) {
+                return $gatewayMap[ $defaultName ];
             }
         }
 
-        return $data;
+        return false;
+    }
+
+    /**
+     * DRY helper to fetch the rename configuration
+     * @return array|bool|\scalar
+     */
+    public function getRenameConfig()
```

Comment:
I'd probably use a more generic config accessor, to be inline with the silverstripe framework. Eg. 
```php
public static function config()
{
    return \Config::inst()->forClass('GatewayFieldsFactory');
}
```

So then you can do: `$this->config()->rename` where you currently use `getRenameConfig`